server/src/main/java/com/google/fhir/gateway/interfaces/AccessDecision.java

```diff
@@ -20,7 +20,17 @@
 
 public interface AccessDecision {
 
-  /** @return true iff access was granted. */
+  /**
+   * Allows the incoming request mutation based on the access decision.
```

Comment:
Since this is an interface, consider elaborating on this comment; e.g., specifying that this is called *after* `canAccess` and only if `canAccess` returns true. Also document that the returned parameters are added to the underlying request before calling the FHIR server.

On that note, I feel `preprocess` is not the best name for this (given that it is called after `canAccess`); how about simply `getRequestMutations`?

Reply:
update the javadoc and renamed to getRequestMutation